geth/api/backend.go

```diff
@@ -208,14 +208,47 @@ func (b *StatusBackend) SendTransaction(ctx context.Context, args common.SendTxA
 	return rst.Hash, nil
 }
 
+func (b *StatusBackend) getVerifiedAccount(password string) (*account.SelectedExtKey, error) {
+	selectedAccount, err := b.accountManager.SelectedAccount()
+	if err != nil {
+		b.log.Error("failed to get a selected account", "err", err)
+		return nil, err
+	}
+	config, err := b.NodeManager().NodeConfig()
+	if err != nil {
+		return nil, err
+	}
+	_, err = b.accountManager.VerifyAccountPassword(config.KeyStoreDir, selectedAccount.Address.String(), password)
+	if err != nil {
+		b.log.Error("failed to verify account", "account", selectedAccount.Address.String(), "error", err)
+		return nil, err
+	}
+	return selectedAccount, nil
+}
+
 // CompleteTransaction instructs backend to complete sending of a given transaction
-func (b *StatusBackend) CompleteTransaction(id common.QueuedTxID, password string) (gethcommon.Hash, error) {
-	return b.txQueueManager.CompleteTransaction(id, password)
+func (b *StatusBackend) CompleteTransaction(id common.QueuedTxID, password string) (hash gethcommon.Hash, err error) {
+	selectedAccount, err := b.getVerifiedAccount(password)
```

Comment:
one more thing, could it be that error is nil and the selected account is nil too? there was a distinct error for such case before, but might be just a legacy

Reply:
`getVerifierAccount` is using [account.SelectedAccount](https://github.com/status-im/status-go/blob/498924f5be57dfea825534873411ca671ba557ad/geth/account/accounts.go#L239) which is returning an error if selectedAccount is nil. So, in the current approach should not possible to have such situation.

However, anybody can directly call tx.CompleteAccount with nil account which will produce a panic [here](https://github.com/status-im/status-go/blob/498924f5be57dfea825534873411ca671ba557ad/geth/transactions/txqueue_manager.go#L175) I'll add a defensive check there and return an `account.ErrNoAccountSelected`. That should keep the same behaviour as before :)

```
	if selectedAccount == nil {
		return account.ErrNoAccountSelected
	}
```